src/molecular-weight.jl

```diff
@@ -0,0 +1,260 @@
+function molecular_weight(aa_seq::AASeq, n_start::Integer, n_end::Integer)
+    AA_vector = Vector{Float64}(undef, 0)
+    i = n_start
+    while i <= n_end
+        if aa_seq[i] == AA_A
+            push!(AA_vector, AA_A_MW)
+        elseif aa_seq[i] == AA_R
+            push!(AA_vector, AA_R_MW)
+        elseif aa_seq[i] == AA_N
+            push!(AA_vector, AA_N_MW)
+        elseif aa_seq[i] == AA_D
+            push!(AA_vector, AA_D_MW)
+        elseif aa_seq[i] == AA_C
+            push!(AA_vector, AA_C_MW)
+        elseif aa_seq[i] == AA_Q
+            push!(AA_vector, AA_Q_MW)
+        elseif aa_seq[i] == AA_E
+            push!(AA_vector, AA_E_MW)
+        elseif aa_seq[i] == AA_G
+            push!(AA_vector, AA_G_MW)
+        elseif aa_seq[i] == AA_H
+            push!(AA_vector, AA_H_MW)  
+        elseif aa_seq[i] == AA_I
+            push!(AA_vector, AA_I_MW)
+        elseif aa_seq[i] == AA_L
+            push!(AA_vector, AA_L_MW)    
+        elseif aa_seq[i] == AA_K
+            push!(AA_vector, AA_K_MW)
+        elseif aa_seq[i] == AA_M
+            push!(AA_vector, AA_M_MW)                                            
+        elseif aa_seq[i] == AA_F
+            push!(AA_vector, AA_F_MW)
+        elseif aa_seq[i] == AA_P
+            push!(AA_vector, AA_P_MW)
+        elseif aa_seq[i] == AA_S
+            push!(AA_vector, AA_S_MW)
+        elseif aa_seq[i] == AA_T
+            push!(AA_vector, AA_T_MW)   
+        elseif aa_seq[i] == AA_W
+            push!(AA_vector, AA_W_MW)
+        elseif aa_seq[i] == AA_Y
+            push!(AA_vector, AA_Y_MW)
+        elseif aa_seq[i] == AA_V
+            push!(AA_vector, AA_V_MW)
+        elseif aa_seq[i] == AA_O
+            push!(AA_vector, AA_O_MW_MW)
+        elseif aa_seq[i] == AA_U
+            push!(AA_vector, AA_U_MW_MW)
+        elseif aa_seq[i] == AA_B
+            push!(AA_vector, AA_B_MW_MW)
+        elseif aa_seq[i] == AA_J
+            push!(AA_vector, AA_J_MW_MW)
+        elseif aa_seq[i] == AA_Z
+            push!(AA_vector, AA_Z_MW_MW)
+        elseif aa_seq[i] == AA_X
+            push!(AA_vector, AA_X_MW_MW)
+        elseif aa_seq[i] == AA_O_Gap
+            push!(AA_vector, AA_O_Gap_MW)                           
+        else
+            println("Error due to unknown amino acid at $i position")
+            return nothing
+        end
+        i = i+1
+    end
+    println("$(round(sum(AA_vector) - ((n_end - n_start - 1) * WATER_MW); digits = 3)) g/mol")
+    println("$(round(((sum(AA_vector) - ((n_end - n_start - 1) * WATER_MW))/1000); digits = 3)) kDa")
+end
+
+# Definning const of molecular weight of monophosphate in g/mol to account for the loss of water and PPi during synthesis
+const DNA_A_MW = 331.22 - WATER_MW          # DNA Adenine                 dAMP ->  Source = https://pubchem.ncbi.nlm.nih.gov/compound/12599
```

Comment:
This is arguable, but instead, I would keep an array of the weights including the water molecule, and then subtract the water weight when computing the sum, just like you do for the amino acids.

Anyway, use two arrays here - one for DNA and one for RNA, just like for amino acids.
This is a little more tricky, because the bitpatterns for A, C, G and T are `0x01`, `0x02`, `0x04` and `0x08`.
But you can use `trailing_zeros(reinterpret(UInt8, dna))` to get 0, 1, 2 and 3, respectively. And then use that to index an array:
```julia
dna_weight = DNA_WEIGHTS[trailing_zeros(reinterpret(UInt8, dna)) + 1]
```

Reply:
Oh I should say - this trick with `trailing_zeros` will not work for ambiguous nucleotides like `DNA_W`, nor will it work for gap (eg. `DNA_Gap`) However, just as with amino acids, you should error if the nucleotide is ambiguous, and set the weight of a gap to zero:
```
dna_weight = if isgap(dna)
    0.0
elseif isambiguous(dna)
    throw(ArgumentError("Cannot compute weight of a nucleotide with ambiguous symbols"))
else
    DNA_WEIGHTS[trailing_zeros(reinterpret(UInt8, dna)) + 1]
end
```